src/m365/purview/commands/auditlog/auditlog-list.ts

```diff
@@ -212,19 +212,17 @@ class PurviewAuditLogListCommand extends O365MgmtCommand {
         logger.logToStderr(`Retrieving content from next ${contentUrisBatch.length} content URIs. Progress: ${Math.round(i / contentUris.length * 100)}%`);
       }
 
-      const batchResult = await Promise.all(
-        contentUrisBatch.map(uri => request.get<any[]>(
-          {
-            url: uri,
-            headers: {
-              accept: 'application/json'
-            },
-            responseType: 'json'
-          }
-        ))
-      );
-
-      batchResult.forEach(res => logs.push(...res));
+      for (const uri of contentUrisBatch) {
+        const response = await request.get<any[]>({
+          url: uri,
+          headers: {
+            accept: 'application/json'
+          },
+          responseType: 'json'
+        });
+
+        logs.push(...response);
+      }
```

Comment:
This is not really refactoring. Right now we are changing the functionality of the code.
Previously we were executing requests in parallel to speed up the command, meanwhile, right now we handle all requests one by one which will increase the execution time quite badly.
Some things just can't be refactored and this is one example, we don't have to get rid of all promises. As far as I know, there's no better alternative for this than to use `Promise.all`.